File: src/test/java/kitchenpos/eatinorders/domain/FakeEatInOrderRepository.java
package kitchenpos.eatinorders.domain;

import kitchenpos.eatinorders.tobe.domain.entity.EatInOrder;
import kitchenpos.eatinorders.tobe.domain.repository.EatInOrderRepository;

import java.util.*;
import java.util.stream.Collectors;

public class FakeEatInOrderRepository implements EatInOrderRepository {
    private Map<UUID, EatInOrder> repository = new HashMap<>();

    @Override
    public EatInOrder save(EatInOrder order) {
        repository.put(order.getId(), order);
        return order;
    }

    @Override
    public Optional<EatInOrder> findById(UUID id) {
        return Optional.ofNullable(repository.get(id));
    }

    @Override
    public List<EatInOrder> findAllByOrderTableId(UUID orderTableId) {
        return repository.values()
                .stream()
                .filter(thisOrder -> thisOrder.getOrderTableId().equals(orderTableId))
                .collect(Collectors.toList());
    }
}

File: src/test/kotlin/kitchenpos/order/tobe/eatinorder/infra/FakeEatInOrderRepository.kt
package kitchenpos.order.tobe.eatinorder.infra

import java.util.*
import kitchenpos.order.tobe.eatinorder.domain.EatInOrder
import kitchenpos.order.tobe.eatinorder.domain.EatInOrderRepository
import kitchenpos.order.tobe.eatinorder.domain.EatInOrderStatus

class FakeEatInOrderRepository(
    private val eatInOrders: MutableMap<UUID, EatInOrder> = mutableMapOf()
) : EatInOrderRepository {
    override fun save(order: EatInOrder): EatInOrder {
        eatInOrders[order.id] = order
        return order
    }

    override fun findById(id: UUID): Optional<EatInOrder> {
        return Optional.ofNullable(eatInOrders[id])
    }

    override fun existsByOrderTableIdAndStatusNot(orderTableId: UUID, orderStatus: EatInOrderStatus): Boolean {
        return eatInOrders.values.any { it.orderTableId == orderTableId && it.status != orderStatus }
    }
}
